test/rekt/features/broker_feature.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"fmt"
+
+	"knative.dev/reconciler-test/pkg/feature"
+
+	"knative.dev/eventing/test/rekt/resources/broker"
+	brokercfg "knative.dev/eventing/test/rekt/resources/broker"
+)
+
+// BrokerGoesReady returns a feature that will create a Broker of the given
+// name and class, and confirm it becomes ready with an address.
+func BrokerGoesReady(name, class string) *feature.Feature {
+	cfg := []brokercfg.CfgFn(nil)
+
+	f := new(feature.Feature)
+
+	// Set the class of the broker.
+	if class != "" {
+		cfg = append(cfg, brokercfg.WithBrokerClass(class))
+	}
+
+	f.Setup(fmt.Sprintf("install broker %q", name), broker.Install(name, cfg...))
+
+	f.Stable("broker").
+		Must("be ready", broker.IsReady(name, interval, timeout)).
+		Must("be addressable", broker.IsAddressable(name, interval, timeout))
```

Comment:
I wonder, would it make sense to have in the context some kind of default value for interval and timeout for all assertions that "retries x times with interval y"? I see no use case where you might want to have different values in different assertions within the same test.

We could have a context decorator that carry those values and then assertions can unwrap them and use in the PollImmediate/WaitForK8sResourceReady

It might be useful for eventshub too

Reply:
I was trying to leave the broker.* lib as dependency and magic value free as possible. so I opted for the timing to be passed down with no magic. Magic an be added for the feature to pull from for sure.

Reply:
It's not a matter of broker.* lib itself, it's really a matter of "how do we handle, across our tests, these 2 values for every assertion we create which are usually fixed for the whole test?". I don't wanna repeat myself too much passing always interval and timeout to any assertion

Reply:
you are not repeating, you are being very direct about they timeout and interval assumptions when calling `broker.IsAddressable` and `broker.IsReady`. There is already a ton of magic happening, I am trying to make these as easy to read as possible

Reply:
I do not feel strongly about this. If you are saying the interval and timeout config is too noisy, I think we can work out a better way.

Reply:
> I do not feel strongly about this. If you are saying the interval and timeout config is too noisy, I think we can work out a better way.

I think it's noisy because it's not related to the _broker is ready concept_. Anyway, let's tackle this in reconciler-test, I can draft a pr for it